geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/IngestCommand.scala

```diff
@@ -238,52 +238,108 @@ object IngestCommand extends LazyLogging {
+        def openStdin(): InputStream = {
+          try {
+            // Get the CachingStdInHandle that gives us access to our custom-made input stream
+            val handle = inputs.handles.headOption.getOrElse {
+              throw new RuntimeException("Unable to obtain handle for stdin")
+            }
 
-      // if there is no converter passed in, try to infer the schema from the input files themselves
-      Command.user.info("No converter defined - will attempt to detect schema from input files")
-      val file = inputs.iterator.flatMap(PathUtils.interpretPath).headOption.getOrElse {
-        throw new ParameterException("Parameter <files> did not evaluate to anything that could be read")
+            // Use the handle to get the stream
+            val (_, stream) = handle.open.next()
+            opened += CloseableIterator.single(stream)
+            System.setIn(stream)
+            System.in
+          } catch {
+            case e: Exception => throw new RuntimeException("Unable to read from stdin", e)
+          }
+        }
+
+        open = openStdin
       }
-      val opened = ListBuffer.empty[CloseableIterator[InputStream]]
-      def open(): InputStream = {
-        val streams = file.open.map(_._2)
-        opened += streams
-        if (streams.hasNext) { streams.next } else {
+      // try to infer the schema from the input files themselves
+      else {
+        Command.user.info("No converter defined - will attempt to detect schema from input files")
+        file = inputs.handles.headOption.getOrElse {
           throw new ParameterException("Parameter <files> did not evaluate to anything that could be read")
         }
+
+        def openFile(): InputStream = {
+          val streams = file.open.map(_._2)
+          opened += streams
+          if (streams.hasNext) {
+            streams.next
+          } else {
+            throw new ParameterException("Parameter <files> did not evaluate to anything that could be read")
+          }
+        }
+
+        open = openFile
       }
+
       val (inferredSft, inferredConverter) = try {
         val opt = format match {
-          case None      => SimpleFeatureConverter.infer(open, Option(sft), Option(file.path))
-          case Some(fmt) => TypeAwareInference.infer(fmt, open, Option(sft), Option(file.path))
+          case None => if (inputs.paths == Inputs.StdInInputs) {
+              SimpleFeatureConverter.infer(open, Option(sft))
+            } else {
+              SimpleFeatureConverter.infer(open, Option(sft), Option(file.path))
+            }
+
+          case Some(fmt) => if (inputs.paths == Inputs.StdInInputs) {
+            TypeAwareInference.infer(fmt, open, Option(sft), None)
+          } else {
+            TypeAwareInference.infer(fmt, open, Option(sft), Option(file.path))
+          }
         }
+
+        // Make stdin point back to the beginning
+        open()
```

Comment:
i think this call shouldn't be necessary, as open will be called during the ingest, which would reset the stream at that point?

Reply:
At first, I didn't have this line, and it was causing the ingest to hang, waiting for input forever. It looks like the ingest is reading from the current position in the stream rather than from the beginning of the stream.

Adding this line had fixed the issue for me.

Reply:
i moved the available checks up to the top of this method, instead of down in the LocalIngestWorker - that way it will validate the available input before any caching.